app/models/exam.rb

```diff
@@ -30,9 +30,13 @@ def enabled_for?(user)
     enabled_range_for(user).cover? DateTime.now
   end
 
-  def accessible_by?(user)
-    #FIXME see with @aguspina
-    user && enabled_for?(user) && authorized?(user)
+  def access!(user)
+    if user.present?
+      raise Exceptions::ExamForbiddenException unless authorized?(user)
+      raise Exceptions::ExamGoneException unless enabled_for?(user)
+    else
+      raise Exceptions::ExamForbiddenException
```

Comment:
If user is not present, shouldn't it be an unauthorized exception, instead of forbidden?